[Bug]: File system support is checked using #ifdef and not #if

### Describe the issue

When building GoogleTest with no file system support (primarily for bare-metal targets), a code section doing filesystem-related operations that should be excluded is compiled anyway, leading to compile and link errors.

### Steps to reproduce the problem

The following example was tested on Ubuntu 20.04 but should have the same behavior on any Linux system.

1- Clone googletest repo in a directory
```bash
git clone https://github.com/google/googletest
```
2- Create the following files next to the googletest folder
  - CMakeLists.txt
```cmake
cmake_minimum_required(VERSION 3.24)
project(GoogleTestNoFilesystem)

add_subdirectory(googletest)
target_compile_definitions(gtest PUBLIC GTEST_HAS_FILE_SYSTEM=0)

add_executable(GoogleTestNoFilesystem main.cpp)
target_link_libraries(GoogleTestNoFilesystem PRIVATE gtest)
```
  - main.cpp
```C++
#include "gtest/gtest.h"

int main(int argc, char* argv[])
{
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}
```
3- Build and run the code
```bash
cmake -S . -B build && cmake --build build
```

Using GCC and GNU Make, similar errors should happen:
```
[ 10%] Building CXX object googletest/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
In file included from googletest/googletest/src/gtest-death-test.cc:90,
                 from googletest/googletest/src/gtest-all.cc:42:
googletest/googletest/src/gtest-internal-inl.h: In member function ‘void testing::internal::UnitTestImpl::AddTestInfo(testing::internal::SetUpTestSuiteFunc, testing::internal::TearDownTestSuiteFunc, testing::TestInfo*)’:
googletest/googletest/src/gtest-internal-inl.h:683:9: error: ‘original_working_dir_’ was not declared in this scope
  683 |     if (original_working_dir_.IsEmpty()) {
      |         ^~~~~~~~~~~~~~~~~~~~~
googletest/googletest/src/gtest-internal-inl.h:684:33: error: ‘FilePath’ has not been declared
  684 |       original_working_dir_.Set(FilePath::GetCurrentDir());
      |                                 ^~~~~~~~
make[2]: *** [googletest/googletest/CMakeFiles/gtest.dir/build.make:76: googletest/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:215: googletest/googletest/CMakeFiles/gtest.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```


### What version of GoogleTest are you using?

40412d85124f7c6f3d88454583c4633e5e10fc8c

### What operating system and version are you using?

Ubuntu 20.04

### What compiler and version are you using?

```bash
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
```

### What build system are you using?

```bash
cmake version 3.25.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

### Additional context

I have created a fix on a fork:
https://github.com/juliencombattelli/googletest/commit/13c309892d1d4c8d692a874d9217f2d238bbd758

Waiting for approval before creating the pull request.

--- a/googletest/src/gtest-internal-inl.h
+++ b/googletest/src/gtest-internal-inl.h
@@ -672,7 +672,7 @@ class GTEST_API_ UnitTestImpl {
   void AddTestInfo(internal::SetUpTestSuiteFunc set_up_tc,
                    internal::TearDownTestSuiteFunc tear_down_tc,
                    TestInfo* test_info) {
-#ifdef GTEST_HAS_FILE_SYSTEM
+#if GTEST_HAS_FILE_SYSTEM
     // In order to support thread-safe death tests, we need to
     // remember the original working directory when the test program
     // was first invoked.  We cannot do this in RUN_ALL_TESTS(), as

